src/set_ref.rs

```diff
@@ -103,22 +103,31 @@ where
     /// Returns `true` if `self` has no elements in common with `other`.
     ///
     /// See also [`HashSet::is_disjoint`].
-    pub fn is_disjoint(&self, other: &HashSet<T, S>) -> bool {
-        self.set.is_disjoint(other, &self.guard)
+    pub fn is_disjoint<Q>(&self, other: &Q) -> bool
+    where
+        Q: AsRef<HashSet<T, S>>,
```

Comment:
What's the reasoning behind making this change? The `HashSet` API in `std` does not have this generic parameter.

Reply:
I made this so that `HashSetRef::is_disjoint` could take either a `&HashSet`or a `&HashSetRef` as an argument.

Reply:
If we end up using two guards in `HashSet::is_disjoint`, we'd have to make `HashSetRef::is_disjoint` take a `HashSetRef`. The same goes for `is_subset` and `is_superset`of course.

Reply:
I'm not sure I follow — why would it need to take `HashSetRef`? The `guard` and `pin` methods should do nicely?

Reply:
> I'm not sure I follow — why would it need to take `HashSetRef`? The `guard` and `pin` methods should do nicely?

Indeed, that would work too. Passing a `HashSetRef` makes more sense to me, as it would allow to user to use different types of guard, but passing a `HashSet`and pinning it works great too.

Reply:
I think I'm actually coming around to you idea of making it generic over `Q: AsRef<HashMap>`. It does mean that we may end up double-pinning, but that shouldn't be too much of an issue.

Reply:
I went for `HashSetRef`in the last commit, as I felt I wasn't able to express my idea without presenting the code. Keep I mind I can always revert back to the gereric type signature if you guys feel like that's a better fit for our API.